### PR TITLE
Eroesler/atm/antarcticarmm

### PR DESCRIPTION
**Pull request by eroesler**
Creates configuration files for atmospheric RRM grid for Antarctica

Creating configuration set-up to run a regionally refined grid (RRM) for the atmospheric component over Antarctica.  The code-changes in files follow the RRM grid for North America.  

[BFB]
[NML]

### CODE REVIEW

**Comment by brhillman** [comment]
The title is a little misleading, this PR appears to add the configuration for both an Antarctic grid *and* a North American grid. I'd suggest modifying the description to reflect that.

File: cime/config/e3sm/config_grids.xml [hunk 1]
```diff
@@ -1806,6 +1829,22 @@
...
+      <file grid="ice|ocn" mask="oRRS15to5">$DIN_LOC_ROOT/share/domains/domain.ocn.northamericax4v1_oRRS15to5.190614.nc</file>
+      <desc>1-deg with 1/4-deg over North America (version 1):</desc>
+    </domain>
+
+    <domain name="ne0np4_antarcticax4v1">
+      <nx>130088</nx>
+      <ny>1</ny>
+      <file grid="atm|lnd" mask="oRRS15to5">$DIN_LOC_ROOT/share/domains/domain.lnd.antarcticax4v1_oRRS15to5.190614.nc</file>
+      <file grid="ice|ocn" mask="oRRS15to5">$DIN_LOC_ROOT/share/domains/domain.ocn.antarcticax4v1_oRRS15to5.190614.nc</file>
+      <desc>1-deg with 1/4-deg over North America (version 1):</desc>
```
Comment (brhillman) [inline cime/config/e3sm/config_grids.xml]: Typo here in the description, should be "Antarctica" instead of "North America".

File: cime/config/e3sm/config_grids.xml [hunk 2]
```diff
@@ -2145,6 +2184,36 @@
       <map name="OCN2ATM_SMAPNAME">cpl/cpl6/map_oRRS18to6_TO_enax4v1_aave.170620.nc</map>
     </gridmap>
 
+    <gridmap atm_grid="ne0np4_northamericax4v1" ocn_grid="oRRS15to5">
+      <map name="ATM2OCN_FMAPNAME">cpl/gridmaps/northamericax4v1np4/map_northamericax4v1_to_oRRS15to5_mono.190613.nc</map>
+      <map name="ATM2OCN_SMAPNAME">cpl/gridmaps/northamericax4v1np4/map_northamericax4v1_to_oRRS15to5_mono.190613.nc</map>
```
Comment (brhillman) [inline cime/config/e3sm/config_grids.xml]: I think the SMAP (mapping weights for state variables) should use the highorder maps, as opposed to the mono maps. Probably the same for the VMAPs.

File: cime/config/e3sm/config_grids.xml [hunk 3]
```diff
@@ -2145,6 +2184,36 @@
...
+      <map name="ATM2OCN_VMAPNAME">cpl/gridmaps/northamericax4v1np4/map_northamericax4v1_to_oRRS15to5_mono.190613.nc</map>
+      <map name="OCN2ATM_FMAPNAME">cpl/gridmaps/oRRS15to5/map_oRRS15to5_to_northamericax4v1_monotr.190613.nc</map>
+      <map name="OCN2ATM_SMAPNAME">cpl/gridmaps/oRRS15to5/map_oRRS15to5_to_northamericax4v1_highorder.190613.nc</map>
+    </gridmap>
+
+    <gridmap atm_grid="ne0np4_northamericax4v1" lnd_grid="360x720cru">
+      <map name="ATM2LND_FMAPNAME">cpl/gridmaps/northamericax4v1np4/map_northamericax4v1_to_360x720cru_mono.190613.nc</map>
+      <map name="ATM2LND_SMAPNAME">cpl/gridmaps/northamericax4v1np4/map_northamericax4v1_to_360x720cru_mono.190613.nc</map>
```
Comment (brhillman) [inline cime/config/e3sm/config_grids.xml]: State map should probably use high order.

File: cime/config/e3sm/config_grids.xml [hunk 4]
```diff
@@ -2145,6 +2184,36 @@
...
+      <map name="LND2ATM_FMAPNAME">cpl/gridmaps/360x720cru/map_360x720cru_to_northamericax4v1_monotr.190613.nc</map>
+      <map name="LND2ATM_SMAPNAME">cpl/gridmaps/360x720cru/map_360x720cru_to_northamericax4v1_monotr.190613.nc</map>
```
Comment (brhillman) [inline cime/config/e3sm/config_grids.xml]: Another state map that should use high order.

File: cime/config/e3sm/config_grids.xml [hunk 5]
```diff
@@ -2145,6 +2184,36 @@
...
+    </gridmap>
+
+    <gridmap atm_grid="ne0np4_antarcticax4v1" ocn_grid="oRRS15to5">
+      <map name="ATM2OCN_FMAPNAME">cpl/gridmaps/antarcticax4v1np4/map_antarcticax4v1_to_oRRS15to5_mono.190613.nc</map>
+      <map name="ATM2OCN_SMAPNAME">cpl/gridmaps/antarcticax4v1np4/map_antarcticax4v1_to_oRRS15to5_mono.190613.nc</map>
```
Comment (brhillman) [inline cime/config/e3sm/config_grids.xml]: State map should be high order.

File: cime/config/e3sm/config_grids.xml [hunk 6]
```diff
@@ -2145,6 +2184,36 @@
...
+      <map name="ATM2OCN_VMAPNAME">cpl/gridmaps/antarcticax4v1np4/map_antarcticax4v1_to_oRRS15to5_mono.190613.nc</map>
+      <map name="OCN2ATM_FMAPNAME">cpl/gridmaps/oRRS15to5/map_oRRS15to5_to_antarcticax4v1_monotr.190613.nc</map>
+      <map name="OCN2ATM_SMAPNAME">cpl/gridmaps/oRRS15to5/map_oRRS15to5_to_antarcticax4v1_highorder.190613.nc</map>
+    </gridmap>
+
+    <gridmap atm_grid="ne0np4_antarcticax4v1" lnd_grid="360x720cru">
+      <map name="ATM2LND_FMAPNAME">cpl/gridmaps/antarcticax4v1np4/map_antarcticax4v1_to_360x720cru_mono.190613.nc</map>
+      <map name="ATM2LND_SMAPNAME">cpl/gridmaps/antarcticax4v1np4/map_antarcticax4v1_to_360x720cru_mono.190613.nc</map>
```
Comment (brhillman) [inline cime/config/e3sm/config_grids.xml]: State map should be high order.

File: cime/config/e3sm/config_grids.xml [hunk 7]
```diff
@@ -2145,6 +2184,36 @@
...
+      <map name="LND2ATM_FMAPNAME">cpl/gridmaps/360x720cru/map_360x720cru_to_antarcticax4v1_monotr.190613.nc</map>
+      <map name="LND2ATM_SMAPNAME">cpl/gridmaps/360x720cru/map_360x720cru_to_antarcticax4v1_monotr.190613.nc</map>
```
Comment (brhillman) [inline cime/config/e3sm/config_grids.xml]: State map should be high order.

File: components/cam/bld/config_files/horiz_grid.xml [hunk 8]
```diff
@@ -42,6 +42,8 @@
 <horiz_grid dyn="se"    hgrid="ne240np4"   ncol="3110402"  csne="240"  csnp="4"  />
 <horiz_grid dyn="se"    hgrid="ne0np4_arm_x8v3_lowcon"   ncol="92558"   csne="0"    csnp="4"  />
 <horiz_grid dyn="se"    hgrid="ne0np4_conus_x4v1_lowcon"   ncol="89147"   csne="0"    csnp="4"  />
+<horiz_grid dyn="se"    hgrid="ne0np4_northamerica_x4v1"   ncol="130088"   csne="0"    csnp="4"  />
```
Comment (brhillman) [inline components/cam/bld/config_files/horiz_grid.xml]: Exact same value of ncol for both the North America and Antarctic grids?! That seems unlikely!

File: components/cam/bld/namelist_files/namelist_defaults_cam.xml [hunk 9]
```diff
@@ -180,6 +182,11 @@
 <ncdata dyn="se" hgrid="ne0np4_conus_x4v1_lowcon"  nlev="30"                     >atm/cam/inic/homme/cami-mam3_0000-01-01_conusx4v1np4_L30_c141106.nc</ncdata>
 <ncdata dyn="se" hgrid="ne0np4_conus_x4v1_lowcon"  nlev="30" ocn="aquaplanet"    >atm/cam/inic/homme/cami_0003-01-01_conusx4v1np4_L30_ape_c000000.nc</ncdata>
 
+<!-- Worked, and used to generate the inic below for northamerica:  ncdata dyn="se" hgrid="ne0np4_northamericax4v1" nlev="72"    >/global/cscratch1/sd/wlin/northamericax4v1/ifs_oper_T1279_2016080100_mod_subset_to_e3sm_northamericax4v1_topoadj_L72.nc</ncdata-->
+<ncdata dyn="se" hgrid="ne0np4_northamericax4v1" nlev="72"    >atm/cam/inic/homme/cami_0001-01-01_northamericax4v1_c190709.nc</ncdata>
```
Comment (brhillman) [inline components/cam/bld/namelist_files/namelist_defaults_cam.xml]: These input files do not appear to be on our inputdata server yet.

File: components/cam/bld/namelist_files/namelist_defaults_cam.xml [hunk 10]
```diff
@@ -223,6 +230,8 @@
 
 <bnd_topo hgrid="ne0np4_arm_x8v3_lowcon"  >atm/cam/inic/homme/USGS-gtopo30_arm_x8v3_lowcon_tensor12xconsistentSGH.nc</bnd_topo>
 <bnd_topo hgrid="ne0np4_conus_x4v1_lowcon"  >atm/cam/topo/USGS_conusx4v1-tensor12x_consistentSGH_c150924.nc</bnd_topo>
+<bnd_topo hgrid="ne0np4_northamericax4v1"  >atm/cam/topo/USGS_northamericax4v1_360x720cru_oRRS15to5_smoothed_190621.nc</bnd_topo>
```
Comment (brhillman) [inline components/cam/bld/namelist_files/namelist_defaults_cam.xml]: Topography files are not on inputdata server.

File: components/cam/bld/namelist_files/namelist_defaults_cam.xml [hunk 11]
```diff
@@ -805,6 +814,8 @@
 <drydep_srf_file hgrid="ne240np4">atm/cam/chem/trop_mam/atmsrf_ne240np4_110920.nc</drydep_srf_file>
 <drydep_srf_file hgrid="ne0np4_arm_x8v3_lowcon">atm/cam/chem/trop_mam/atmsrf_armx8v3.nc</drydep_srf_file>
 <drydep_srf_file hgrid="ne0np4_conus_x4v1_lowcon">atm/cam/chem/trop_mam/atmsrf_conusx4v1.nc</drydep_srf_file>
+<drydep_srf_file hgrid="ne0np4_northamericax4v1">atm/cam/chem/trop_mam/atmsrf_northamericax4v1np4_190627.nc</drydep_srf_file>
```
Comment (brhillman) [inline components/cam/bld/namelist_files/namelist_defaults_cam.xml]: Dry deposition files are not on inputdata server.

File: components/cam/bld/namelist_files/namelist_defaults_cam.xml [hunk 12]
```diff
@@ -1392,6 +1403,8 @@
 <mesh_file>/dev/null</mesh_file>
 <mesh_file hgrid="ne0np4_arm_x8v3_lowcon"  >atm/cam/inic/homme/arm_x8v3_lowcon.g</mesh_file>
 <mesh_file hgrid="ne0np4_conus_x4v1_lowcon"  >atm/cam/inic/homme/conusx4v1.g</mesh_file>
+<mesh_file hgrid="ne0np4_northamericax4v1"  >atm/cam/inic/homme/northamericax4v1.g</mesh_file>
```
Comment (brhillman) [inline components/cam/bld/namelist_files/namelist_defaults_cam.xml]: Mesh files are not on inputdata server.

File: components/cam/bld/namelist_files/namelist_defaults_cam.xml [hunk 13]
```diff
@@ -1410,17 +1423,14 @@
 <nu hgrid="ne240np4">  1.1e12 </nu>
 <nu hgrid="ne0np4_arm_x8v3_lowcon"  > 8.0e-8</nu>
 <nu hgrid="ne0np4_conus_x4v1_lowcon"  > 8.0e-8</nu>
+<nu hgrid="ne0np4_northamericax4v1"  > 8.0e-8</nu>
```
Comment (brhillman) [inline components/cam/bld/namelist_files/namelist_defaults_cam.xml]: This is nit-picky, but lining up the right brackets would make the values easier to read off here (as is done above for the "uniform" resolution grids).

File: components/cam/bld/namelist_files/namelist_defaults_cam.xml [hunk 14]
```diff
@@ -1700,6 +1722,7 @@
 <fatmlndfrc hgrid="ne120np4"  >share/domains/domain.lnd.ne120np4_gx1v6.111226.nc</fatmlndfrc>
 <fatmlndfrc hgrid="ne240np4"  >share/domains/domain.lnd.ne240np4_gx1v6.120125.nc</fatmlndfrc>
 
+<!--fatmlndfrc hgrid="ne0np4_northamericax4v1">share/domains/domain.lnd.northamericax4v1_oRRS15to5.190614.nc</fatmlndfrc-->
```
Comment (brhillman) [inline components/cam/bld/namelist_files/namelist_defaults_cam.xml]: Why add this commented out line?

File: components/cam/bld/namelist_files/namelist_defaults_cam.xml [hunk 15]
```diff
@@ -1792,6 +1815,7 @@
 <focndomain hgrid="ne120np4"  >share/domains/domain.ocn.ne120np4_gx1v6.111226.nc</focndomain>
 <focndomain hgrid="ne240np4"  >share/domains/domain.ocn.ne240np4_gx1v6.111226.nc</focndomain>
 
+<!--focndomain hgrid="ne0np4_northamericax4v1">share/domains/domain.ocn.northamericax4v1_oRRS15to5.190614.nc</focndomain-->
```
Comment (brhillman) [inline components/cam/bld/namelist_files/namelist_defaults_cam.xml]: As above...why is this here?

**Comment by rljacob** [comment]
@eroesler you need to respond to reviewer comments.

**Comment by wlin7** [comment]
This branch can be updated to address the issues Ben commented on regarding the Antarcttica grid, after the northamerica grid in a separate PR (#3067) is merged.  

@eroesler , can you please respond to Ben's comments? You may opt to update PR 3067 first.

**Comment by rljacob** [comment]
@eroesler you have comments to respond too.

**Comment by tangq** [comment]
Rebased to current master.

**Comment by rljacob** [comment]
@tangq will you be getting this PR ready to merge?

**Comment by tangq** [comment]
I am working on this PR.

I see there are many conflicting files (probably because of the North America RRM merged yesterday). @rljacob , shall we resolve these conflicts? Or start a new branch from latest master?

**Comment by wlin7** [comment]
@tangq, it is ok to continue using this branch. The conflicts must be on the northamerica entries themselves, which are now duplicated in this PR. In that case, fixing the conflicts with rebasing should be straightforward.

**Comment by rljacob** [comment]
@tangq what is the status of this PR?

**Comment by tangq** [comment]
This PR was replaced by this branch https://github.com/E3SM-Project/E3SM/tree/tangq/atm/antarcticarrm, because addressing the review comments in this PR will take longer time than starting a new branch.

I will issue another PR for the Antarctica RRM based on the new branch above. It seems fine to close this PR now.
